File: MuizzaApp1/MuizzaApp1/Models/PurchaseResult.cs
namespace MuizzaApp1.Models
{
    public class PurchaseResult
    {
        public bool Success { get; }
        public string ErrorMessage { get; }
        public string TransactionId { get; }

        private PurchaseResult(bool success, string errorMessage = null, string transactionId = null)
        {
            Success = success;
            ErrorMessage = errorMessage;
            TransactionId = transactionId;
        }

        public static PurchaseResult Successful(string transactionId = null) => new PurchaseResult(true, transactionId: transactionId);
        public static PurchaseResult Failed(string errorMessage) => new PurchaseResult(false, errorMessage);
    }
} 

File: MuizzaApp/MuizzaApp/Models/PurchaseResult.swift
import Foundation

struct PurchaseResult {
    let success: Bool
    let errorMessage: String?
    let transactionId: String?

    static func successful(transactionId: String? = nil) -> PurchaseResult {
        PurchaseResult(success: true, errorMessage: nil, transactionId: transactionId)
    }

    static func failed(_ errorMessage: String) -> PurchaseResult {
        PurchaseResult(success: false, errorMessage: errorMessage, transactionId: nil)
    }
}
